app/src/main/java/com/babylon/wallet/android/presentation/settings/preferences/gateways/GatewaysScreen.kt

```diff
@@ -77,51 +74,57 @@ fun GatewaysScreen(
         modifier = modifier,
         state = state,
         onBackClick = onBackClick,
-        onAddGatewayClick = viewModel::onAddGateway,
-        onNewUrlChanged = viewModel::onNewUrlChanged,
         onDeleteGateway = viewModel::onDeleteGateway,
         onGatewayClick = viewModel::onGatewayClick,
+        onAddGatewayClick = { viewModel.setAddGatewaySheetVisible(true) },
         oneOffEvent = viewModel.oneOffEvent,
-        onCreateProfile = onCreateProfile,
-        addGatewaySheetVisible = viewModel::setAddGatewaySheetVisible
+        onCreateProfile = onCreateProfile
+    )
+
+    val bottomSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
+    SyncSheetState(
+        sheetState = bottomSheetState,
+        isSheetVisible = state.addGatewayInput != null,
+        onSheetClosed = { viewModel.setAddGatewaySheetVisible(false) }
     )
+
+    if (state.addGatewayInput != null) {
+        DefaultModalSheetLayout(
+            sheetState = bottomSheetState,
+            showDragHandle = true,
+            wrapContent = true,
+            onDismissRequest = { viewModel.setAddGatewaySheetVisible(false) },
+            sheetContent = {
+                state.addGatewayInput?.let {
```

Comment:
do we need that line here, given that few lines above we check if addGatewayInput is not null?

Reply:
Nope, you're right, thanks